server/src/cache/pool.rs

```diff
@@ -0,0 +1,94 @@
+use deadpool_redis::{redis, Pool, Runtime};
+use snafu::prelude::Snafu;
+
+const REDIS_POOL_SIZE: usize = 100;
+
+#[derive(Clone)]
+#[allow(missing_debug_implementations)]
+pub struct RedisPool {
+    pool: Pool, // doesn't implement Debug
+    expiration_ms: u32,
+}
+
+#[derive(Debug, Clone, Copy, Snafu)]
+pub enum CacheError {
+    #[snafu(display("Could not build configuration for cache."))]
+    CouldNotConfigure,
+
+    #[snafu(display("Could not connect to redis pool."))]
+    CouldNotConnect,
+
+    #[snafu(display("The operation on the redis cache failed."))]
+    OperationFailed,
+}
+
+impl RedisPool {
+    pub async fn new(
+        config: crate::config::Config,
+        expiration_ms: u32,
+    ) -> Result<Self, CacheError> {
+        cache_info!("(RedisPool new) entry.");
+        let port = config.docker_port_redis;
+        let host = config.docker_host_redis;
+
+        let cfg = deadpool_redis::Config::from_url(format!("redis://{host}:{port}"));
```

Comment:
A nice thing about using the Config module, is that you can directly get a `deadpool_redis::Config` object out of it when using the `__`  seperator.
An example is given here: https://docs.rs/deadpool-redis/latest/deadpool_redis/struct.Config.html

So, if we would define the following env vars in the `.env` file:
```
REDIS__CONNECTION__ADDR=redis:6479
REDIS__POOL__MAX_SIZE=100
```

And then add the `redis` key to our `Config` struct:
```
/// struct holding configuration options
#[derive(Debug, Deserialize, Clone)]
pub struct Config {
    /// port to be used for gRPC server
    pub docker_port_grpc: u16,
    /// port to be used for REST server
    pub docker_port_rest: u16,
    /// url to be used for the Redis server
    redis: deadpool_redis::Config,
    /// path to log configuration YAML file
    pub log_config: String,
}
```

You can shorten the function body to:
```
        cache_info!("(RedisPool new) entry.");
        match config.redis.create_pool(Some(Runtime::Tokio1)) {
            Ok(pool) => Ok(RedisPool {
                pool,
                expiration_ms,
            }),
            Err(e) => {
                cache_error!("Could not create new redis pool from config: {}", error);
                Err(CacheError::CouldNotConfigure)
            }
        }
```

Reply:
https://github.com/Arrow-air/svc-telemetry/pull/15/files/0f4d2b06942616bfc605c6c219e6824d3cbd8b66..1ae030ec8c5fc5bf790b7d78560d28c78cdbf4af